### PR TITLE
feat: add dataset descriptor parameters to automate generation of title and description TDE-955

### PR DESCRIPTION
**Pull request by MDavidson17**
#### Motivation

Entering/typing a title or description for imagery datasets is currently inefficient and risky. To remove the risk the title & description should be automatically generated using input parameters.

#### Modification

imagery-standardising workflow updated to take: `gsd`, `subtype`, `location`, `event`, `historic-survey-number` and `lifecycle` parameters.

create-collection step updated to take above parameters and `region`, `start-datetime` and `end-datetime`

standardise-publish-import workflow updated
ascii-standardise-publish workflow updated

#### Checklist

_If not applicable, provide explanation of why._

- [x] Tests updated - no tests
- [x] Docs updated
- [x] Issue linked in Title

Should not be merged before https://github.com/linz/topo-imagery/pull/757 is merged & released

### CODE REVIEW

File: workflows/raster/standardising.yaml [hunk 1]
```diff
@@ -102,10 +102,19 @@ spec:
...
+        enum:
+          - 'Aerial Photos'
+          - 'Urban Aerial Photos'
+          - 'Rural Aerial Photos'
+          - 'Scanned Aerial Imagery'
+          - 'DEM'
+          - 'DSM'
+          - 'Satellite Imagery'
+      - name: gsd
+        description: 'Dataset GSD. e.g.- 0.3m'
```
Comment (l0b0) [inline workflows/raster/standardising.yaml:116]: I don't think that hyphen is grammatical.

We might want to quote the example value.

File: workflows/raster/standardising.yaml [hunk 2]
```diff
@@ -238,6 +247,23 @@ spec:
...
+        description: '(Optional) Additional datatset description, to be used in the title in place of the Region, e.g.- Hamilton 0.05m Urban Aerial Photos (2023)'
+        value: ''
+      - name: lifecycle
+        value: 'completed'
+        enum:
+          - 'under development'
+          - 'preview'
+          - 'ongoing'
+          - 'completed'
+          - 'deprecated'
```
Comment (l0b0) [inline workflows/raster/standardising.yaml:260]: Is there a way we could generate this from [the reference](https://github.com/linz/stac/blob/e39b6b78780aeb976a52e7e5086f7fc316c3d21b/extensions/linz/schema.json#L986-L986)?

File: workflows/raster/standardising.yaml (hunk 1 above)
Comment (MDavidson17) [reply workflows/raster/standardising.yaml]: I have removed the default value, the GSD is a required parameter for generating the title

File: workflows/raster/standardising.yaml (hunk 2 above)
Comment (MDavidson17) [reply workflows/raster/standardising.yaml:259]: That would be nice, I dont think there is a simple way to do that.

Comment (l0b0) [reply workflows/raster/standardising.yaml:259]: I guess not. Maybe later.

**Review by l0b0 — DISMISSED** [review]

File: workflows/raster/README.md [hunk 3]
```diff
@@ -273,16 +281,20 @@ This workflow carries out the steps in the [Standardising](#Standardising) workf
...
+| source-epsg            | str   | 2193         | The EPSG code of the source imagery.                                                                                                                                                                                                             |
+| target-epsg            | str   | 2193         | The Target EPSG code, if different to source-epsg the imagery will be reprojected.                                                                                                                                                               |
+| location               | str   | ''           | (Optional) Additional datatset description, to be used in dataset title / description in place of the Region.                                                                                                                                    |
+| lifecycle              | enum  | Completed    | Lifecycle Status of Collection                                                                                                                                                                                                                   |
```
Comment (paulfouquet) [inline workflows/raster/README.md:295]:
```suggestion
| lifecycle              | enum  | completed    | Lifecycle Status of Collection                                                                                                                                                                                                                   |
```

**Review by l0b0 — APPROVED** [review]

**Review by paulfouquet — APPROVED** [review]

**Review by blacha — APPROVED** [review]

**Pull request merged**